c/tskit/tables.c

```diff
@@ -6701,7 +6701,11 @@ tsk_table_collection_check_mutation_integrity(
                 goto out;
             }
         }
-        /* Check known/unknown times are not both present */
+
+        /* Check known/unknown times are not both present on a site*/
+        if (j > 0 && mutations.site[j - 1] != mutations.site[j]) {
+            unknown_times_seen = false;
```

Comment:
Should probably do this below where we set ``last_known_time = INFINITY;``. We're inherently depending on the site ordering being correct anyway, so we might as will avoid the extra comparison.

Reply:
I had it there initially - but then the check can only happen if `check_mutation_ordering` is set.

Reply:
We'll only catch mixed stuff at a site if all the mutations for a given site are adjacent, so we depend on this ordering being true anyway.

Reply:
Cool, I've moved it and changed the test.